crates/kitsune_p2p/fetch/src/respond.rs

```diff
@@ -54,20 +54,24 @@ impl<C: FetchResponseConfig> FetchResponseQueue<C> {
     }
 
     /// Enqueue an op to be sent to a remote.
-    pub fn enqueue_op(&self, space: KSpace, user: C::User, op: KOpData) {
+    pub fn enqueue_op(&self, space: KSpace, user: C::User, op: KOpData) -> bool {
         let len = op.size();
 
-        if len > u32::MAX as usize {
-            tracing::error!("op size > u32::MAX");
-            return;
+        // Don't try to take more permits than the byte_limit has. That would panic!
+        if len > self.config.byte_limit() as usize {
```

Comment:
This wasn't testable as it was, it meant allocating a lot of memory. The actual upper bound on what can be accepted is much lower than u32::MAX because the semaphore will error if asked to give out more semaphores than it holds (not current, but max). This is easier to test and sufficient to prevent unexpected errors in the next step

Reply:
Actually this check isn't a hard requirement. It just means we'd get a confusing error about the fetch responder being overloaded rather than a correct message about the op size being too large. Probably worth keeping even if the tests pass without it

Reply:
If we want to be pedantic, this limit is also bound to [`usize::MAX >> 3`](https://docs.rs/tokio/1.31.0/tokio/sync/struct.Semaphore.html#associatedconstant.MAX_PERMITS), which could be significant on a 32 bit system.

Reply:
The docs are saying we'd get a panic if we tried to construct a semaphore that is over that size so that's covered, then checking a lower value than that when handing out permits should be safe right?

Reply:
Ah, good point :+1: